src/common.cc

```diff
@@ -111,4 +121,343 @@ Join(const std::vector<std::string>& vec, const std::string& delim)
   return ss.str();
 }
 
+#ifdef TRITON_BIG_ENDIAN
+
+/// Returns uint16_t input with byte order swapped
+///
+/// \param[in] input 
```

Comment:
I like this `\param[in]` syntax! I think it would be especially useful and clarifying for our functions that use `\param[in,out]` . Not in this PR, but I think we should update our API docs to use this syntax.